packages/core/src/lib/filterPeers.spec.ts

```diff
@@ -1,16 +1,19 @@
+import { generateKeyPair } from "@libp2p/crypto/keys";
 import { Peer } from "@libp2p/interface";
 import type { Tag } from "@libp2p/interface";
-import { createSecp256k1PeerId } from "@libp2p/peer-id-factory";
+import { peerIdFromPrivateKey } from "@libp2p/peer-id";
 import { Tags } from "@waku/interfaces";
 import { expect } from "chai";
 
 import { filterPeersByDiscovery } from "./filterPeers.js";
 
 describe("filterPeersByDiscovery function", function () {
   it("should return all peers when numPeers is 0", async function () {
-    const peer1 = await createSecp256k1PeerId();
-    const peer2 = await createSecp256k1PeerId();
-    const peer3 = await createSecp256k1PeerId();
+    const [peer1, peer2, peer3] = await Promise.all([
+      generateKeyPair("secp256k1").then(peerIdFromPrivateKey),
```

Comment:
it seems to be repetitive thing, let's extract into `tests/utils`

Reply:
we often also require the `privateKey` so fine to do it this way